include/net/nrf_cloud_codec.h

```diff
@@ -125,6 +127,40 @@ struct nrf_cloud_obj_shadow_accepted {
 	struct nrf_cloud_obj config;
 };
 
+struct nrf_cloud_obj_shadow_transform_result {
+	/** The object containing the result of the transform request.
+	 *  The result will have a key of @ref NRF_CLOUD_TRANSFORM_RSP_TF_KEY.
+	 *  The result value type/format depends on the transform request and can be a JSON object,
+	 *  string, number, or boolean.
+	 */
+	struct nrf_cloud_obj obj;
+};
```

Comment:
I'm not sold on this needing to be its own struct, but I don't feel super strongly

Reply:
otherwise LGTM